File: Leetcode_9.java
public class Leetcode_9 {
    public boolean isPalindrome(int x) {
        long reversed= 0;
        int original = x;
        if(x < 0){ return false; }
        if(x < 10){ return true; }
        while(x > 0){
            reversed= reversed * 10 + x%10;
            x= x/10;
        }
        return reversed== original ;
    }
}

File: python/Leetcode_9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        re_num= 0
        original= x
        if x < 0:
            return False
        if x <10:
            return True
        while x > 0:
            re_num= re_num * 10 + x%10
            x= x//10
        return re_num == original